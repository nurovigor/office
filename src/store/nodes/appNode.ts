import { node } from 'redux-nodes';
import { Nullable } from 'src/common/types/types';

export const appNode = node(
	{
		//initialState
		error: null as Nullable<string>
	},
	{
		//actions
		setError: (state, error: Nullable<string>) => (state.error = error)
	}
);
