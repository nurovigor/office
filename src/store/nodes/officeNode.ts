import { node } from 'redux-nodes';
import { Nullable } from 'src/common/types/types';

export const officeNode = node(
	{
		ids: [] as string[],
		tables: {} as TableType,
		developers: [] as DeveloperType[]
	},
	{
		//actions
		setIds: (state, ids: string[]) => (state.ids = ids),
		setTables: (state, tables: TableType) => (state.tables = tables),
		setDevs: (state, devs: DeveloperType[]) => (state.developers = devs),
		setUpdateTable: (state, updatedTable: TableTypeI) =>
			(state.tables[updatedTable._id] = updatedTable),
		removeDevOldTable: (state, tableId: string) => (state.tables[tableId].developer = null)
	}
);

export interface TableTypeI {
	camera: string;
	developer: Nullable<DeveloperType>;
	headphones: string;
	keyboard: string;
	microphone: string;
	monitor: string;
	mouse: string;
	number: string;
	openSpaceNumber: string;
	pc: string;
	__v: number;
	_id: string;
}

export type DeveloperType = {
	_id: string;
	firstName: string;
	lastName: string;
	phone: string;
	_v: number;
};

export type TableType = {
	[key: string]: TableTypeI;
};
