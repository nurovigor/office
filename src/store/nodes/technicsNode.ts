import { node } from 'redux-nodes';
import { FilterName, FilterType, Nullable, TechnicTypeI } from 'src/common/types/types';

export const technicsNode = node(
	{
		//initialState
		ids: [] as string[],
		technics: [] as TechnicTypeI[],
		currentPage: 1 as number,
		totalCountItems: 0 as number,
		pageSize: 10,
		filter: [] as FilterType[],
		name: '' as string,
		type: '' as string,
		bind: '' as string,
		sort: [] as string[],
		selectedItem: null as Nullable<string>,
		isFetching: false
	},
	{
		//actions
		setIds: (state, ids: string[]) => (state.ids = ids),
		setTechnics: (state, technics: TechnicTypeI[]) => (state.technics = technics),
		setTotalCount: (state, count: number) => (state.totalCountItems = count),
		setCurrentPage: (state, currentPage: number) => (state.currentPage = currentPage),
		setFilter: (state, filterName: string, filter: string) =>
			(state[filterName as keyof FilterName] = filter),
		setFilterOptions: (state, filterOptions: FilterType[]) => (state.filter = filterOptions),
		setOption: (state, filterName: string, filter: string) =>
			(state.filter = state.filter.map((item) =>
				item.name === filterName ? { ...item, selectedOption: filter } : item
			)),
		setSort: (state, name: string, value: string) => (state.sort = [name, value]),
		setItem: (state, id: Nullable<string>) => (state.selectedItem = id),
		removeItem: (state, id) => (state.technics = state.technics.filter((item) => item._id !== id)),
		updateItem: (state, id: string, updatedItem: TechnicTypeI) =>
			(state.technics = state.technics.map((item) => (item._id === id ? updatedItem : item))),
		addNewItem: (state, newItem: TechnicTypeI) => (state.technics = [...state.technics, newItem]),
		setIsFetching: (state, status: boolean) => (state.isFetching = status)
	}
);
