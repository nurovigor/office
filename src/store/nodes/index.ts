import { buildNodes } from 'redux-nodes';
import { appNode } from './appNode';
import { officeNode } from './officeNode';
import { technicsNode } from './technicsNode';

export default buildNodes({
	appNode,
	officeNode,
	technicsNode
});
