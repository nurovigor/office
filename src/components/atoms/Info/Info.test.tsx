import React from 'react';
import { render } from '@testing-library/react';
import { Info } from './';
import { TableTypeI } from 'src/store/nodes/officeNode';

describe('Info component', () => {
	let tableData: TableTypeI;
	let count: (keyof TableTypeI)[];

	beforeEach(() => {
		tableData = {
			camera: 'Canon',
			developer: {
				_id: '1',
				firstName: 'Front',
				lastName: 'End',
				phone: '+79889999999',
				_v: 1
			},
			headphones: 'Panasonic',
			keyboard: 'A4Tech',
			microphone: 'Sven',
			monitor: 'ViewSonic',
			mouse: 'Logitech',
			number: '1',
			openSpaceNumber: '1',
			pc: 'Evo',
			__v: 1,
			_id: '1'
		};
		count = Object.keys(tableData) as (keyof TableTypeI)[];
	});
	it('renders Info component', () => {
		const { asFragment } = render(<Info count={count} table={tableData} />);
		expect(asFragment()).toMatchSnapshot();
	});
	it('renders Info without developer', () => {
		tableData = {
			camera: 'Canon',
			developer: null,
			headphones: 'Panasonic',
			keyboard: 'A4Tech',
			microphone: 'Sven',
			monitor: 'ViewSonic',
			mouse: 'Logitech',
			number: '1',
			openSpaceNumber: '1',
			pc: 'Evo',
			__v: 1,
			_id: '1'
		};
		const { asFragment } = render(<Info count={count} table={tableData} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
