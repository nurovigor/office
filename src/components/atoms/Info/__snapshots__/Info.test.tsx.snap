// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component renders Info component 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        developer:
      </div>
      <div>
        Front End
      </div>
    </div>
    <div>
      <div>
        camera:
      </div>
      <div>
        Canon
      </div>
    </div>
    <div>
      <div>
        developer:
      </div>
      <div>
        [object Object]
      </div>
    </div>
    <div>
      <div>
        headphones:
      </div>
      <div>
        Panasonic
      </div>
    </div>
    <div>
      <div>
        keyboard:
      </div>
      <div>
        A4Tech
      </div>
    </div>
    <div>
      <div>
        microphone:
      </div>
      <div>
        Sven
      </div>
    </div>
    <div>
      <div>
        monitor:
      </div>
      <div>
        ViewSonic
      </div>
    </div>
    <div>
      <div>
        mouse:
      </div>
      <div>
        Logitech
      </div>
    </div>
    <div>
      <div>
        number:
      </div>
      <div>
        1
      </div>
    </div>
    <div>
      <div>
        openSpaceNumber:
      </div>
      <div>
        1
      </div>
    </div>
    <div>
      <div>
        pc:
      </div>
      <div>
        Evo
      </div>
    </div>
    <div>
      <div>
        __v:
      </div>
      <div>
        1
      </div>
    </div>
    <div>
      <div>
        _id:
      </div>
      <div>
        1
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Info component renders Info without developer 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        developer:
      </div>
      <div>
        empty
      </div>
    </div>
    <div>
      <div>
        camera:
      </div>
      <div>
        Canon
      </div>
    </div>
    <div>
      <div>
        developer:
      </div>
      <div>
        null
      </div>
    </div>
    <div>
      <div>
        headphones:
      </div>
      <div>
        Panasonic
      </div>
    </div>
    <div>
      <div>
        keyboard:
      </div>
      <div>
        A4Tech
      </div>
    </div>
    <div>
      <div>
        microphone:
      </div>
      <div>
        Sven
      </div>
    </div>
    <div>
      <div>
        monitor:
      </div>
      <div>
        ViewSonic
      </div>
    </div>
    <div>
      <div>
        mouse:
      </div>
      <div>
        Logitech
      </div>
    </div>
    <div>
      <div>
        number:
      </div>
      <div>
        1
      </div>
    </div>
    <div>
      <div>
        openSpaceNumber:
      </div>
      <div>
        1
      </div>
    </div>
    <div>
      <div>
        pc:
      </div>
      <div>
        Evo
      </div>
    </div>
    <div>
      <div>
        __v:
      </div>
      <div>
        1
      </div>
    </div>
    <div>
      <div>
        _id:
      </div>
      <div>
        1
      </div>
    </div>
  </div>
</DocumentFragment>
`;
